xblock/query.py

```diff
@@ -0,0 +1,226 @@
+"""
+    Define Query and Shared objects. Query supports ad-hoc queries via get and set operators.
+    Shared uses `bind` functions to access data from pre-defined entities.  
+"""
+
+from xblock.exceptions import SharedFieldAccessDeniedError
+
+class Query(object):
+    """ 
+    A query object for accessing shared field in XBlock with Queryable operators
+
+    Parameters:
+
+        field_name (str): the name of the shared field
+    """
+    def __init__(self, field_name):
+        self._queryable = Queryable(field_name=field_name)
```

Comment:
`Query` is going to be instantiated on a per-xblock-class basis, right? Which means that `_queryable` will be as well. I worry about having the "current" xblock attached to a class-global object.

Could you instead instantiate a new `Queryable` in `__get__`?